src/objects/bullet.gd

```diff
@@ -41,26 +41,30 @@ func set_target(pos: Vector2):
 
 
 func _physics_process(delta):
-	if spawned_by == "demon":
+	if spawned_by.get_name().to_lower() == "demon":
```

Comment:
🤔 Yah I think we're reaching some sort of limit on doing this sort of `stringly type checking`.

Recommendation:

We lean into duck typing and call the methods we want on the object itself.